Data import & preparation

1. Historical population dataset (csv) includes a range of annual labour market statistics and indicators from 1960 broken
2. Regional social and environmental indicators dataset (csv) includes around 40 indicators of demography, economic accounts, labour market, social and innovation themes
3. Green growth indicators dataset (csv) indicators for monitoring progress towards green growth
4. Economic Outlook dataset (API) covers annual and quarterly data from 1960 until 2023


In [6]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [7]:
green_growth = pd.read_csv("GreenGrowth_OECD.csv", index_col = 0)
green_growth.head()

Country         VAR                                           Variable  \
COU                                                                             
AUS  Australia  CO2_PBPROD  Production-based CO2 productivity, GDP per uni...   
AUS  Australia  CO2_PBPROD  Production-based CO2 productivity, GDP per uni...   
AUS  Australia  CO2_PBPROD  Production-based CO2 productivity, GDP per uni...   
AUS  Australia  CO2_PBPROD  Production-based CO2 productivity, GDP per uni...   
AUS  Australia  CO2_PBPROD  Production-based CO2 productivity, GDP per uni...   

      YEA  Year Unit Code                     Unit  PowerCode Code PowerCode  \
COU                                                                            
AUS  1990  1990    USD_KG  US dollars per kilogram               0     Units   
AUS  1995  1995    USD_KG  US dollars per kilogram               0     Units   
AUS  2000  2000    USD_KG  US dollars per kilogram               0     Units   
AUS  2010  2010    USD_KG  US dollars per kilogram               0     Units   
AUS  2015  2015    USD_KG  US dollars per kilogram               0     Units   

    Reference Period Code Reference Period     Value Flag Codes Flags  
COU                                                                    
AUS                  2015             2015  1.961928        NaN   NaN  
AUS                  2015             2015  2.091676        NaN   NaN  
AUS                  2015             2015  2.156981        NaN   NaN  
AUS                  2015             2015  2.556194        NaN   NaN  
AUS                  2015             2015  3.020283        NaN   NaN

In [8]:
green_growth.describe()

YEA           Year  PowerCode Code         Value
count  143260.000000  143260.000000   143260.000000  1.432600e+05
mean     2010.670739    2010.670739        0.181223  4.011952e+06
std         9.876944       9.876944        1.026890  1.053445e+09
min      1990.000000    1990.000000        0.000000 -5.059876e+03
25%      2000.000000    2000.000000        0.000000  1.276220e+00
50%      2016.000000    2016.000000        0.000000  1.189200e+01
75%      2018.000000    2018.000000        0.000000  6.515363e+01
max      2021.000000    2021.000000        6.000000  3.217270e+11

In [9]:
green_growth.columns

Index(['Country', 'VAR', 'Variable', 'YEA', 'Year', 'Unit Code', 'Unit',
       'PowerCode Code', 'PowerCode', 'Reference Period Code',
       'Reference Period', 'Value', 'Flag Codes', 'Flags'],
      dtype='object')

In [10]:
green_growth.info()

<class 'pandas.core.frame.DataFrame'>
Index: 143260 entries, AUS to MHL
Data columns (total 14 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   Country                143260 non-null  object 
 1   VAR                    143260 non-null  object 
 2   Variable               143260 non-null  object 
 3   YEA                    143260 non-null  int64  
 4   Year                   143260 non-null  int64  
 5   Unit Code              129641 non-null  object 
 6   Unit                   129641 non-null  object 
 7   PowerCode Code         143260 non-null  int64  
 8   PowerCode              143260 non-null  object 
 9   Reference Period Code  21206 non-null   object 
 10  Reference Period       21206 non-null   object 
 11  Value                  143260 non-null  float64
 12  Flag Codes             26069 non-null   object 
 13  Flags                  26069 non-null   object 
dtypes: float64(1), int64(3), object(10)
memory

In [11]:
green_growth['Variable']

COU
AUS    Production-based CO2 productivity, GDP per uni...
AUS    Production-based CO2 productivity, GDP per uni...
AUS    Production-based CO2 productivity, GDP per uni...
AUS    Production-based CO2 productivity, GDP per uni...
AUS    Production-based CO2 productivity, GDP per uni...
                             ...                        
MHL                                    Biomass, % of DMC
MHL                                    Biomass, % of DMC
MHL                                    Biomass, % of DMC
MHL                                    Biomass, % of DMC
MHL                                    Biomass, % of DMC
Name: Variable, Length: 143260, dtype: object

In [12]:
green_growth.Variable.value_counts()

Population density, inhabitants per km2                     2437
Population                                                  2300
Annual surface temperature, change since 1951-1980          2243
Nominal exchange rate                                       2236
GDP deflator                                                2170
                                                            ... 
Threatened bird species, % total known species                71
Threatened vascular plant species, % total known species      69
Threatened mammal species, % total known species              67
Total renewable freshwater per capita                         42
Environmentally related R&D expenditure, % GDP                39
Name: Variable, Length: 159, dtype: int64

In [13]:
green_growth.Country.value_counts()

Denmark              1182
Japan                1180
France               1177
Netherlands          1171
Norway               1165
                     ... 
Saint Martin           57
Saint Barthélemy       51
Antarctica             42
Clipperton Island      22
Bonaire                22
Name: Country, Length: 237, dtype: int64

In [14]:
unique_countries = green_growth["Country"].unique()
print(unique_countries)

['Australia' 'Austria' 'Belgium' 'Canada' 'Czech Republic' 'Denmark'
 'Finland' 'France' 'Germany' 'Greece' 'Hungary' 'Iceland' 'Ireland'
 'Italy' 'Japan' 'Korea' 'Luxembourg' 'Mexico' 'Netherlands' 'New Zealand'
 'Norway' 'Poland' 'Portugal' 'Slovak Republic' 'Spain' 'Sweden'
 'Switzerland' 'Turkey' 'United Kingdom' 'United States' 'Brazil' 'Chile'
 "China (People's Republic of)" 'Colombia' 'Estonia' 'India' 'Indonesia'
 'Israel' 'Latvia' 'Russia' 'Slovenia' 'South Africa' 'World'
 'BRIICS economies - Brazil, Russia, India, Indonesia, China and South Africa'
 'OECD - Total' 'OECD - Europe' 'OECD America' 'OECD Asia Oceania'
 'Saudi Arabia' 'Argentina' 'Costa Rica' 'Lithuania' 'Iran' 'Jordan'
 'Brunei Darussalam' 'Turkmenistan' 'Azerbaijan' "Côte d'Ivoire" 'Algeria'
 'Philippines' 'Viet Nam' 'Egypt' 'Thailand' 'Bolivia'
 'Dominican Republic' 'Panama' 'Peru' 'Lebanon' 'Nicaragua' 'Morocco'
 'Ukraine' 'Haiti' 'Bulgaria' 'Croatia' 'Guatemala' 'Cambodia' 'Paraguay'
 'Iraq' 'Georgia' 'Tunis

In [15]:
unique_variables = green_growth["Variable"].unique()
print(unique_variables)

['Production-based CO2 productivity, GDP per unit of energy-related CO2 emissions'
 'Production-based CO2 intensity, energy-related CO2 per capita'
 'Energy productivity,  GDP per unit of TPES'
 'Energy intensity, TPES per capita'
 'Renewable energy supply, % total energy supply'
 'Renewable electricity, % total electricity generation'
 'Total freshwater abstraction per capita'
 'Water stress, total freshwater abstraction as % total available renewable resources'
 'Water stress, total freshwater abstraction as % total internal renewable resources'
 'Threatened mammal species, % total known species'
 'Threatened bird species, % total known species'
 'Threatened vascular plant species, % total known species'
 'Population connected to public sewerage, % total population'
 'Environmentally related government R&D budget, % total government R&D'
 'Environmentally related R&D expenditure, % GDP'
 'Renewable energy public RD&D budget, % total energy public RD&D'
 'Energy public RD&D budget, % 

In [16]:
grouped_variables=green_growth.groupby(['Country','Variable']).count()['Value'].sort_values(ascending=False)
print(grouped_variables)

Country     Variable                                                                        
Bhutan      Real GDP, Index 2000=100                                                            11
Spain       Terrestrial protected area, % land area                                             11
Costa Rica  Marine protected area, % total exclusive economic zone                              11
Tanzania    Annual surface temperature, change since 1951-1980                                  11
Belgium     Marine protected area, % total exclusive economic zone                              11
                                                                                                ..
Cyprus      Petrol tax, USD per litre                                                            1
            Seasonal surface water, % total surface                                              1
Mali        Conversion of seasonal to permanent water surface, % permanent water, since 1984     1
Turkey      Conv

In [46]:
green_growth_significant=green_growth.groupby(['Country','Variable']).filter(lambda v: (v.Value.size >= 11))

In [47]:
green_growth_significant.head()

Country           VAR  \
COU                            
AUS  Australia  ENVRD_GBAORD   
AUS  Australia  ENVRD_GBAORD   
AUS  Australia  ENVRD_GBAORD   
AUS  Australia  ENVRD_GBAORD   
AUS  Australia  ENVRD_GBAORD   

                                              Variable   YEA  Year Unit Code  \
COU                                                                            
AUS  Environmentally related government R&D budget,...  1990  1990        PC   
AUS  Environmentally related government R&D budget,...  1995  1995        PC   
AUS  Environmentally related government R&D budget,...  2000  2000        PC   
AUS  Environmentally related government R&D budget,...  2010  2010        PC   
AUS  Environmentally related government R&D budget,...  2015  2015        PC   

           Unit  PowerCode Code PowerCode Reference Period Code  \
COU                                                               
AUS  Percentage               0     Units                   NaN   
AUS  Percentage               0     Units                   NaN   
AUS  Percentage               0     Units                   NaN   
AUS  Percentage               0     Units                   NaN   
AUS  Percentage               0     Units                   NaN   

    Reference Period     Value Flag Codes  Flags  
COU                                               
AUS              NaN  3.115022          B  Break  
AUS              NaN  1.194272          B  Break  
AUS              NaN  2.950042          B  Break  
AUS              NaN  3.953611          B  Break  
AUS              NaN  3.828087          B  Break

In [48]:
unique_variables_significant = green_growth_significant["Variable"].unique()
print(unique_variables_significant)

['Environmentally related government R&D budget, % total government R&D'
 'Real GDP per capita' 'Population density, inhabitants per km2'
 'Renewable energy public RD&D budget, % total energy public RD&D'
 'Energy public RD&D budget, % GDP'
 'Value added in services, % of total value added'
 'Real GDP, Index 2000=100' 'Diesel tax, USD per litre'
 'Petrol tax, USD per litre'
 'Value added in industry, % of total value added'
 'Value added in agriculture, % of total value added'
 'Fossil fuel public RD&D budget (excluding CCS), % total energy public RD&D'
 'Diesel end-user price, USD per litre'
 'Petrol end-user price, USD per litre'
 'Industry electricity price, USD per kWh'
 'Residential electricity price, USD per kWh'
 'Annual surface temperature, change since 1951-1980'
 'Terrestrial protected area, % land area'
 'Marine protected area, % total exclusive economic zone'
 'Farmland Birds Index, 2000=100' 'Irrigated land, % agricultural land'
 'GDP deflator' 'Nominal exchange rate' 'Pur

In [54]:
main_variables=green_growth_significant.groupby(['Variable']).count()['Value'].sort_values(ascending=False)
print(main_variables)

Variable
Population density, inhabitants per km2                                       2013
GDP deflator                                                                  1936
Nominal exchange rate                                                         1914
Purchasing power parity                                                       1837
Annual surface temperature, change since 1951-1980                            1815
Terrestrial protected area, % land area                                       1221
Marine protected area, % total exclusive economic zone                        1045
Real GDP, Index 2000=100                                                       990
Real GDP per capita                                                            990
Real GDP                                                                       990
Value added in industry, % of total value added                                264
Value added in agriculture, % of total value added                            

Picking 3 variables on area:
Terrestrial protected area, % land area                                       1221
Marine protected area, % total exclusive economic zone                        1045
Irrigated land, % agricultural land                                             22

Also 1 variable on population density:
Population density, inhabitants per km2                                       2013

In [73]:
variable_set = set(['Terrestrial protected area, % land area','Marine protected area, % total exclusive economic zone','Irrigated land, % agricultural land','Population density, inhabitants per km2'])


In [74]:
sliced_green_growth = green_growth_significant[green_growth_significant['Variable'] in variable_set]                                               

TypeError: 'Series' objects are mutable, thus they cannot be hashed